### PR TITLE
[docs] Update AuditLogDiff.type to be accurate

### PR DESCRIPTION
**Pull request by Artemis21**
### Summary

The documentation for `AuditLogDiff.type` is currently inaccurate and incomplete. This pull request fixes that, using data from [the Discord Developer Docs](https://discord.com/developers/docs/resources/channel#channel-object-channel-types).

### Checklist

- [x] If code changes were made then they have been tested (none have been made).
    - [N/A] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -2170,8 +2170,23 @@ this goal, it must make use of a couple of data classes that aid in this goal.
 
         The type of channel or channel permission overwrite.
 
-        If the type is an :class:`int`, then it is a type of channel which can be either
-        ``0`` to indicate a text channel or ``1`` to indicate a voice channel.
+        If the type is an :class:`int`, then it is one of the following:
+
+        +---+---------------------------------+
+        | 0 | A guild text channel.           |
```
Comment (Gobot1234) [inline docs/api.rst:2176]: I think these digits should still be wrapped in ``

File: docs/api.rst [hunk 2]
```diff
@@ -2170,8 +2170,23 @@ this goal, it must make use of a couple of data classes that aid in this goal.
 
         The type of channel or channel permission overwrite.
 
-        If the type is an :class:`int`, then it is a type of channel which can be either
-        ``0`` to indicate a text channel or ``1`` to indicate a voice channel.
+        If the type is an :class:`int`, then it is one of the following:
+
+        +---+---------------------------------+
```
Comment (NCPlayz) [inline docs/api.rst:2175]:
```suggestion
        +-------+---------------------------------+
        | value |     meaning                     |
        +-------+---------------------------------+
```
seems to make sense to have a table heading similar to this.

File: docs/api.rst (hunk 1 above)
Comment (Artemis21) [reply docs/api.rst]: Sure, done

File: docs/api.rst (hunk 2 above)
Comment (Artemis21) [reply docs/api.rst]: Accidentally outdated that but I will add.

File: docs/api.rst [hunk 3]
```diff
@@ -2170,8 +2170,23 @@ this goal, it must make use of a couple of data classes that aid in this goal.
 
         The type of channel or channel permission overwrite.
 
-        If the type is an :class:`int`, then it is a type of channel which can be either
-        ``0`` to indicate a text channel or ``1`` to indicate a voice channel.
+        If the type is an :class:`int`, then it is one of the following:
+
+        +-----+---------------------------------+
+        | `0` | A guild text channel.           |
```
Comment (Gobot1234) [inline docs/api.rst:2176]: I meant like \``0``

File: docs/api.rst (hunk 2 above)
Comment (Artemis21) [reply docs/api.rst]: Ok done, and as a header row too

File: docs/api.rst (hunk 3 above)
Comment (Artemis21) [reply docs/api.rst]: Oh right sorry I am not familiar with rst

**Comment by Rapptz** [comment]
This is already encapsulated and referenced under the ``ChannelType`` enum so re-typing it there feels... duplication heavy. I don't really like this approach because of it.

**Comment by Artemis21** [comment]
> This is already encapsulated and referenced under the `ChannelType` enum so re-typing it there feels... duplication heavy. I don't really like this approach because of it.

True, though the documentation enum doesn't include the numbers, which are necessary. I suppose it would be nice to cast `AuditLogDiff.type` to an enum in the library, but that would be a breaking change.

**Comment by Rapptz** [comment]
Well the easiest way would be to reference the numbers inside the enum documentation and then reference the enum within the AuditLogDiff.type

**Comment by Artemis21** [comment]
The someone harder, but nicer (and still backwards compatible) way would be to cast `AuditLogDiff.type` to an `IntEnum` or equivalent.

**Comment by Rapptz** [comment]
I don't use the `enum` module so that suggestion is out regardless.

**Comment by Artemis21** [comment]
So I see, but an equivalent could work I think, by adding a `__int__` method around line 60 of `enums.py`...

**Comment by Artemis21** [comment]
Actually, I think that would be too much of a breaking change - since the value can currently be either an `int` or a `str`, people will likely be checking if `isinstance(AuditLogDiff.type, int)`, which would be `False` - for it not to be breaking, it would have to actually inherit from `int` like `IntEnum` does.

**Pull request closed without merging**